### DISCUSSION TITLE
Image upload doesnt work

### DISCUSSION THREAD

**Discussion by jsegal** [mastodon/mastodon#19568] [General]
Hello, I am new to using this and so far when I try to upload any image to a post it says processing but never completes the upload.

**Comment by megmaker** [comment]
Am also new, and also experiencing this. Images are in proper file format and under the size limit. System takes a long while to upload even small images, then hangs at a "processing..." message.

**Reply by BenRossTransit** [reply]
Am also new, and also having same problem.

**Comment by Gargron** [comment]
Whichever server you signed up (probably mastodon.social in this case) is overloaded.

Mastodon has to convert the files you upload into proper media formats and create thumbnails. It doesn't do that within your upload request, but in a background processing queue. This background processing queue can get backlogged (more jobs come in than can be processed at the same time) and then individual jobs can get delayed by a long time. This affects all sorts of things but specifically here it affects the ability to process uploaded files. Technically if you waited for 52 minutes the processing would actually complete (I realize nobody is going to wait that long).

This has occurred 3 times during the last 3 days and I spent a lot of time fixing it (it seems that by CET midnight, the incoming activity subsides enough that the processing queue can finally catch up to real time). Currently I am waiting for my hosting provider to provision more hardware I ordered. I apologize for this bad experience and want to assure you that Mastodon is not usually like this -- usually it's very snappy and fast and supports all kinds of media uploads.

You could try signing up on a different Mastodon server that isn't currently overloaded to get a feel for what it's like for it to be working properly. Servers are just service providers that host your account--no matter which one you're on, you can follow and talk to anyone else on the network.

**Reply by megmaker** [reply]
@Gargron Many thanks for your kind reply, and lengthy explanation. I suspected the server was getting overloaded because of the influx of people (like me!) joining and trying it out. I'll be patient. I do appreciate your efforts. Cheers!

**Reply by nigini** [reply]
Hey, @Gargron : I am maintaining a Mastodon server that got to this issue lately. Do you have any documentation about where to focus energy/hardware to tackle this media upload issue?

**Reply by MaryMuse** [reply]
I am also having this issue. (Images upload, preview shows, when I save it goes to an error page). I see the files on my S3 install. Hoping to find a direction for troubleshooting and/or help. Thanks.

**Comment by jsegal** [comment]
Thank you! Twitter must have massive server farms of unimaginable size to
manage this. Thank you for your efforts.  I will try smaller servers.
<http://youtu.be/3VtaVXZMu-s>

___________________________________
*“The simple act of caring is heroic.” – Edward Albert*

Thank you,
Joseph Segal
Author
JosephSegal.com

**Comment by jsegal** [comment]
Can a Mastodon site server be run on a VPS server?
How much is it costing you to run yours if you don't mind sharing?
Is it nonprofit or do you accept donations or how do you mange the expense?

Thank you!
<http://youtu.be/3VtaVXZMu-s>

___________________________________
*“The simple act of caring is heroic.” – Edward Albert*

Thank you,
Joseph Segal
Author
JosephSegal.com
